docker/dev/Dockerfile

```diff
@@ -1,60 +1,80 @@
-FROM ubuntu:20.04
+FROM ubuntu:22.04
 
 # tzdata package is installed implicitly in the following command. This package
 # sets timezone interactively during the installation process. This environment
 # variable prevents this interaction.
 ARG DEBIAN_FRONTEND=noninteractive
+ARG INSTALL_COMMAND="apt-get -y install --no-install-recommends"
+
+# Copy prebuilt packages.
+COPY docker/packages /packages/
 
 # CodeCompass development dependencies.
-RUN set -x && apt-get update -qq \
-  && apt-get -y install --no-install-recommends \
-  cmake make \
-  default-jdk \
-  ctags \
-  ca-certificates \
-  curl \
-  gnupg \
-  doxygen \
-  gcc-9 gcc-9-plugin-dev g++-9 \
+RUN set -x \
```

Comment:
In Docker, each `RUN` statement creates a separate [layer](https://docs.docker.com/build/guide/layers/). So if e.g. in one layer you add a big file to an image, and in the next layer you delete it, the final image size will be big, because a layer contains that file, regardless that it was removed later.

Therefore it was a deliberate decision to install all packages from the Ubuntu package repository in a *single step* and then in the *same* step remove the APT caches (`apt-get clean` and removing `/var/lib/apt/lists/`). This way the image will be significantly smaller, because the caches won't be added to any of the layers.

With your refactoring into separate `RUN` command and thus layers, this will not like this anymore, so I suggest to revert this.